src/main/drivers/flash_m25p16.c

```diff
@@ -243,20 +251,32 @@ bool m25p16_init(const flashConfig_t *flashConfig)
     return m25p16_readIdentification();
 }
 
+void m25p16_setAddress(bool useLongAddress, uint32_t address, uint8_t *buf)
+{
+    if (useLongAddress) {
+        *buf++ = (address >> 24) & 0xff;
+    }
+    *buf++ = (address >> 16) & 0xff;
+    *buf++ = (address >> 8) & 0xff;
+    *buf = address & 0xff;
+}
+
 /**
  * Erase a sector full of bytes to all 1's at the given byte offset in the flash chip.
  */
 void m25p16_eraseSector(uint32_t address)
```

Comment:
Following routines may be modeled as something line:
```
void m25p16_StartAddressCmd(uint8_t cmd, uint32_t address, bool writeEnable)
{
  uint8_t buff[5];
  uint8_t *p = buff;
  *p++ = cmd;
  if (isLargeFlash) {
    *p++ = (address >> 24) & 0xff;
  }
  *p++ = (address >> 16) & 0xff;
  *p++ = (address >> 8) & 0xff;
  *p++ = address & 0xff;

  if (!m25p16_waitForReady(DEFAULT_TIMEOUT_MILLIS))
    return 0;
  if(writeEnable) {   // maybe length > 0
    m25p16_writeEnable();
  } 
  ENABLE_M25P16;
  spiTransfer(bus->busdev_u.spi.instance, out, NULL, p - buffer);
}

void m25p16_eraseSector(uint32_t address)
{
   m25p16_StartAddressCmd(M25P16_INSTRUCTION_SECTOR_ERASE , address, true)
   DISABLE_M25P16;
}

```

Reply:
@ledvinap 
I'll keep the current code for now, as I'm planning to change the begin-continue-end code not to monopolize the bus.